go/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"periph.io/x/host/v3"
+)
+
+const (
+	LEDPinName    = "18"
+	ButtonPinName = "23"
+)
+
+func main() {
+	var err error
+
+	_, err = host.Init()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	printer := NewPrinter()
+
+	// we have our context so we can terminate the program
```

Comment:
A go context is something that you pass to async operations so that they can timeout or cancel, etc. https://blog.golang.org/context